### DISCUSSION TITLE
Can we handle the Embedded documents efficiently.

### DISCUSSION THREAD

**Discussion by harshalizode** [dymmond/mongoz#45] [Q&A]
Hello Everyone,
I want to perform the CRUD operations on the Embedded documents.
* like fetching the embedded document by its specific field and update or delete it.
* Creating indexes for the embedded documents, so that we ensure the same document is not getting inserted in the embedded document.
* Also, we can add here the implementation like the Django reverse query?

**Comment by tarsil** [comment]
@harshalizode thank you and we do understand that but https://mongoz.tarsild.io/embedded-documents/?h=em is not design for that. That is the job of a Document. 

Embedded documents are not documents and do not have CRUD operations.

**Reply by tarsil** [reply]
Handling "efficiently" means nothing in this case. If we convert the embedded document to perform CRUD, it will be.... A document. Makes absolutely no sense.

**Reply by harshalizode** [reply]
Can we then retrieve the embedded document by filtering specific fields in the Document without iterating over the embedded documents?

**Reply by tarsil** [reply]
The embedded document if you see it is a simple pydantic base model with Mongoz fields. I don't see how would that would be possible? You simply query the document and you can select only the field corresponding to the embedded document? Although that could be a nice possible feature to add later

**Reply by harshalizode** [reply]
With the aggregation query it is possible to  filter the nested embedded documents. Can we implement this in our Mongoz Query Manager?
